### PR TITLE
fix (and test!) GitHub provider resolution

### PR DESCRIPTION
**Pull request by minrk**
fixes failure introduced in #264

### CODE REVIEW

File: binderhub/repoproviders.py [hunk 1]
```diff
@@ -136,7 +136,7 @@ def _default_auth(self):
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self.user, self.repo, self.unresolved_ref = self.process_spec(self.spec)
```
Comment (betatim) [inline binderhub/repoproviders.py]: So the bug was calling a method that didn't exist anymore?

Comment (minrk) [reply binderhub/repoproviders.py]: Yup. Last commit moved method to top-level function, but didn't keep up with the rename.

**Comment by minrk** [comment]
Thanks! Will deploy to staging as soon as it's uploaded.

File: binderhub/repoproviders.py (hunk 1 above)
Comment (hydrosquall) [reply binderhub/repoproviders.py]: Oops, my bad! Sorry about that, thanks for catching this

Comment (minrk) [reply binderhub/repoproviders.py]: @hydrosquall no problem at all! That's why we have tests (now). Thanks for your help setting that up.

**Pull request merged**